compiler/ast/ast.go

```diff
@@ -303,7 +303,7 @@ type TupleExpr struct {
 	Loc   `json:"loc"`
 }
 
-type QueryExpr struct {
+type Subquery struct {
```

Comment:
Can we call this SubqueryExpr to make clear that it's an expression?

Reply:
At it's core it's an expression that wraps an operator so really both at the same time.  In the runtime, it implements both Evaluator and Op.    I realize in the AST it just implements exprNode but I think the bridge concept is clear enough that subquery can stand alone.

Reply:
I can live with Subquery.

But while I get that in the runtime a subquery implements both Evaluator and Puller, in the AST and DAG they're strictly expressions. Yes, they wrap operators, but they aren't operators themselves!